### ISSUE TITLE
Auto codec negotiation not working with `v0.1-rc.3` under specific situation

### ISSUE THREAD

**Issue by felipecrs** [AlexxIT/go2rtc#108]
PS: It used to work in `v0.1-rc.2`.

I cannot get audio in WebRTC for `camera.rua` with the following:

```yaml
streams:
  rua_hd_aac:
    - ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
  camera.rua:
    - rtsp://admin:12345@192.168.1.33:554/0/av1
    - rtsp://localhost:8554/rua_hd_aac?audio=aac
```

But with the following it works:

```yaml
streams:
  rua_hd_aac:
    - ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
  camera.rua:
    - rtsp://admin:12345@192.168.1.33:554/0/av1
    # With the line below commented
    # - rtsp://localhost:8554/rua_hd_aac?audio=aac
```

Of course the drawback is that MSE player can't play audio anymore since the AAC track won't be present any longer in camera.rua.

@AlexxIT I think this is probably a regression, codecs auto-negotiation not working properly (it's not selecting the PCM/8000 for the WebRTC consumer where it can).

_Originally posted by @felipecrs in https://github.com/AlexxIT/go2rtc/issues/65#issuecomment-1313781635_

Logs:

```log
14:37:33.299 INF go2rtc version 0.1-rc.3 linux/amd64
14:37:33.300 INF [api] listen addr=:1984
14:37:33.300 INF [rtsp] listen addr=:8554
14:37:33.302 INF [webrtc] listen addr=:8555
14:37:33.302 INF [srtp] listen addr=:8443
14:37:49.837 TRC [streams] consumer=0 candidate=0 media="video, recvonly"
14:37:49.837 DBG [streams] probe producer url=rtsp://admin:12345@192.168.1.33:554/0/av1
14:37:49.846 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:37:49.846 TRC [streams] match producer:0:0 => consumer:0:0 codec="96 H264/90000"
14:37:49.849 TRC [streams] consumer=0 candidate=1 media="audio, recvonly, 0 MPEG4-GENERIC/0"
14:37:49.849 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:37:49.849 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 8 PCMA/8000"
14:37:49.849 DBG [streams] probe producer url=rtsp://localhost:8554/rua_hd_aac?audio=copy
14:37:49.850 TRC [streams] consumer=0 candidate=0 media="audio, recvonly"
14:37:49.850 DBG [streams] probe producer url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:37:49.860 TRC [streams] consumer=0 candidate=0 media="video, recvonly"
14:37:52.609 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:37:52.610 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:37:52.610 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:37:52.610 TRC [streams] match producer:0:0 => consumer:0:0 codec="96 H264/90000"
14:37:52.610 TRC [streams] match producer:0:1 => consumer:0:0 codec="97 MPEG4-GENERIC/8000"
14:37:52.610 TRC [streams] consumer=0 candidate=1 media="audio, recvonly"
14:37:52.610 DBG [streams] start producer url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:37:52.610 TRC [streams] producer=1 candidate=0 media="audio, sendonly, 96 MPEG4-GENERIC/8000"
14:37:52.610 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:37:52.610 TRC [streams] match producer:1:0 => consumer:0:1 codec="96 MPEG4-GENERIC/8000"
14:37:52.610 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:37:52.610 TRC [streams] match producer:0:1 => consumer:0:1 codec="97 MPEG4-GENERIC/8000"
14:37:52.610 DBG [streams] start producer url=rtsp://admin:12345@192.168.1.33:554/0/av1
14:37:52.610 DBG [streams] start producer url=rtsp://localhost:8554/rua_hd_aac?audio=copy
14:38:01.530 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:105 > error="read tcp [::1]:59776->[::1]:8554: i/o timeout"
14:38:01.530 DBG [streams] reconnect to url=rtsp://localhost:8554/rua_hd_aac?audio=copy
14:38:01.531 TRC [streams] consumer=1 candidate=0 media="audio, recvonly"
14:38:01.531 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:01.531 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:38:01.531 TRC [streams] match producer:0:1 => consumer:1:0 codec="97 MPEG4-GENERIC/8000"
14:38:06.532 DBG github.com/AlexxIT/go2rtc/cmd/streams/producer.go:153 > error="read tcp [::1]:33190->[::1]:8554: i/o timeout"
14:38:06.532 DBG [streams] reconnect to url=rtsp://localhost:8554/rua_hd_aac?audio=copy
14:38:06.533 TRC [streams] consumer=1 candidate=0 media="audio, recvonly"
14:38:06.533 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:06.533 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:38:06.533 TRC [streams] match producer:0:1 => consumer:1:0 codec="97 MPEG4-GENERIC/8000"
14:38:09.875 TRC [streams] consumer=1 candidate=0 media="video, recvonly"
14:38:09.876 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:09.876 TRC [streams] match producer:0:0 => consumer:1:0 codec="96 H264/90000"
14:38:09.876 TRC [streams] consumer=1 candidate=1 media="audio, recvonly"
14:38:09.876 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:09.876 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:38:09.876 TRC [streams] match producer:0:1 => consumer:1:1 codec="97 MPEG4-GENERIC/8000"
14:38:10.226 TRC [streams] consumer=1 candidate=0 media="video, recvonly, 96 VP8/90000, 97 RTX/90000, 98 VP9/90000, 99 RTX/90000, 100 VP9/90000, 101 RTX/90000, 35 VP9/90000, 36 RTX/90000, 37 VP9/90000, 38 RTX/90000, 102 H264/90000, 122 RTX/90000, 127 H264/90000, 121 RTX/90000, 125 H264/90000, 107 RTX/90000, 108 H264/90000, 109 RTX/90000, 124 H264/90000, 120 RTX/90000, 39 H264/90000, 40 RTX/90000, 41 H264/90000, 42 RTX/90000, 43 H264/90000, 44 RTX/90000, 45 AV1/90000, 46 RTX/90000, 47 AV1/90000, 48 RTX/90000, 123 H264/90000, 119 RTX/90000, 114 RED/90000, 115 RTX/90000, 116 ULPFEC/90000, 49 FLEXFEC-03/90000"
14:38:10.226 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:10.226 TRC [streams] match producer:0:0 => consumer:1:0 codec="96 H264/90000"
14:38:10.226 TRC [streams] consumer=1 candidate=1 media="audio, recvonly, 111 OPUS/48000/2, 63 RED/48000/2, 103 ISAC/16000, 104 ISAC/32000, 9 G722/8000, 0 PCMU/8000, 8 PCMA/8000, 106 CN/32000, 105 CN/16000, 13 CN/8000, 110 TELEPHONE-EVENT/48000, 112 TELEPHONE-EVENT/32000, 113 TELEPHONE-EVENT/16000, 126 TELEPHONE-EVENT/8000"
14:38:10.227 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:10.227 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 8 PCMA/8000"
14:38:10.227 TRC [streams] match producer:0:1 => consumer:1:1 codec="8 PCMA/8000"
14:38:10.227 WRN [stream] can't get track
14:38:10.227 TRC [streams] producer=1 candidate=0 media="audio, sendonly, 96 MPEG4-GENERIC/8000"
14:38:10.227 TRC [streams] consumer=1 candidate=2 media="audio, sendonly, 111 OPUS/48000/2, 63 RED/48000/2, 103 ISAC/16000, 104 ISAC/32000, 9 G722/8000, 0 PCMU/8000, 8 PCMA/8000, 106 CN/32000, 105 CN/16000, 13 CN/8000, 110 TELEPHONE-EVENT/48000, 112 TELEPHONE-EVENT/32000, 113 TELEPHONE-EVENT/16000, 126 TELEPHONE-EVENT/8000"
14:38:10.227 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:10.227 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 8 PCMA/8000"
14:38:10.227 TRC [streams] producer=1 candidate=0 media="audio, sendonly, 96 MPEG4-GENERIC/8000"
TODO: webrtc ontrack &{mu:{w:{state:0 sema:0} writerSem:0 readerSem:0 readerCount:0 readerWait:0} id:10685791-4819-4a4b-acda-5b88f9899f96 streamID:lnFKYe1aX1calZyzDZXb9mUEAy6q3ITyAAij payloadType:111 kind:1 ssrc:237034584 codec:{RTPCodecCapability:{MimeType:audio/opus ClockRate:48000 Channels:2 SDPFmtpLine:minptime=10;useinbandfec=1 RTCPFeedback:[{Type:transport-cc Parameter:}]} PayloadType:111 statsID:} params:{HeaderExtensions:[{URI:http://www.ietf.org/id/draft-holmer-rmcat-transport-wide-cc-extensions-01 ID:4}] Codecs:[{RTPCodecCapability:{MimeType:audio/opus ClockRate:48000 Channels:2 SDPFmtpLine:minptime=10;useinbandfec=1 RTCPFeedback:[{Type:transport-cc Parameter:}]} PayloadType:111 statsID:}]} rid: receiver:0xc0000bc300 peeked:[144 239 126 89 152 250 134 143 14 32 220 88 190 222 0 1 65 0 1 0 120 128 210 166 175 49 154 13 154 200 213 60 45 134 55 110 93 6 177 232 105 200 123 5 27 235 203 54 198 243 152 140 87 150 98 62 105 217 98 7 137 48 55 23 172 64 188 1 97 5 26 208 202 24 172 53 155 113 177 253 133 85 185 11 204 234 158 11 72 215 0 211 115 116 118] peekedAttributes:map[0:0xc000289810]}
14:38:11.530 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:105 > error="read tcp [::1]:8554->[::1]:59792: i/o timeout"
14:38:11.531 DBG [streams] reconnect to url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:38:11.535 DBG github.com/AlexxIT/go2rtc/cmd/streams/producer.go:153 > error="read tcp [::1]:33206->[::1]:8554: i/o timeout"
14:38:11.535 DBG [streams] reconnect to url=rtsp://localhost:8554/rua_hd_aac?audio=copy
14:38:11.535 TRC [streams] consumer=1 candidate=0 media="audio, recvonly"
14:38:14.888 ERR [api.ws] readJSON error="websocket: close 1000 (normal)"
14:38:20.129 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:20.129 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:38:20.129 TRC [streams] match producer:0:1 => consumer:1:0 codec="97 MPEG4-GENERIC/8000"
14:38:20.129 DBG [streams] stop producer url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:38:20.129 WRN [stream] can't get track
14:38:20.129 DBG [streams] can't stop none producer
14:38:20.129 TRC [streams] stop reconnect url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:38:20.129 WRN [rtsp] error="couldn't find the matching tracks" stream=rua_hd_aac
14:38:20.130 TRC [streams] consumer=0 candidate=0 media="audio, recvonly"
14:38:20.130 DBG [streams] probe producer url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
14:38:22.864 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
14:38:22.864 TRC [streams] producer=0 candidate=1 media="audio, sendonly, 97 MPEG4-GENERIC/8000"
14:38:22.864 TRC [streams] match producer:0:1 => consumer:0:0 codec="97 MPEG4-GENERIC/8000"
14:38:22.864 DBG [streams] start producer url=ffmpeg:rtsp://admin:12345@192.168.1.33:554/0/av0#video=copy#audio=aac
```

**Comment by AlexxIT** [comment]
I don't like line
```
14:37:52.609 TRC [streams] producer=0 candidate=0 media="video, sendonly, 96 H264/90000"
```
It shouldn't double. You need to increase log level for other modules.

**Comment by AlexxIT** [comment]
A very complicated situation. I think it has something about running `rua_hd_aac` from two different places at the same time. I can't repeat it and I can't figure out where the error is.

**Comment by felipecrs** [comment]
Ok I have more information to provide: the issue only happens when Frigate is consuming the stream.

If I turn off Frigate, then it works normally. So I guess this has something to do when there's already one or two consumers and another one tries to connect.

**Comment by felipecrs** [comment]
This also seems to be fixed in v0.1-rc.5. Thanks a lot, @AlexxIT!
